hash.c

```diff
@@ -98,6 +98,12 @@ rb_hash_set_ifnone(VALUE hash, VALUE ifnone)
     return hash;
 }
 
+VALUE
+rb_hash_resurrect(VALUE hash)
```

Comment:
why don't you make new function?
(why don't you use rb_hash_dup directly?)

Reply:
Because we have `rb_ary_resurrect` and `rb_str_resurrect`. I can change to use `rb_hash_dup` directly.